crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -2743,19 +2743,13 @@ impl TwoQubitControlledUDecomposer {
         gates1.global_phase = global_phase;
         Ok(gates1)
     }
-}
 
-#[pymethods]
-impl TwoQubitControlledUDecomposer {
-    ///  Initialize the KAK decomposition.
-    ///  Args:
-    ///      rxx_equivalent_gate: Gate that is locally equivalent to an :class:`.RXXGate`:
-    ///      :math:`U \sim U_d(\alpha, 0, 0) \sim \text{Ctrl-U}` gate.
-    ///  Raises:
-    ///      QiskitError: If the gate is not locally equivalent to an :class:`.RXXGate`.
-    #[new]
-    #[pyo3(signature=(rxx_equivalent_gate))]
-    pub fn new(py: Python, rxx_equivalent_gate: RXXEquivalent) -> PyResult<Self> {
+    /// Initialize the KAK decomposition.
+    pub fn new_inner(
+        py: Python,
+        rxx_equivalent_gate: RXXEquivalent,
+        euler_basis: &str,
+    ) -> PyResult<Self> {
         let atol = DEFAULT_ATOL;
```

Comment:
I know we discussed the addition of a `new_inner` class, but the current implementation doesn't seem to add (or remove) anything to the existing `new` class, so I would just keep and use `new` in the rust code.

Reply:
the `new` function is `pyo3` method -- isn't this function faster? if not, I can change it back.

Reply:
I managed to remove the need of `py` in [231af7f](https://github.com/Qiskit/qiskit/pull/13568/commits/231af7f18b4ce6b69d82c526d1c3d2488c899f91)
by replacing it with gil, as suggested by Matthew in https://github.com/Qiskit/qiskit/pull/13419.
I still think we need both functions `new` and `new_inner` since one of them is in pure rust and the other is pyo3.

Reply:
Yes, let's keep them.